src/lib.rs

```diff
@@ -1,497 +1,442 @@
-    "l" => "1",
-    "m" => "9",
-    "n" => "4",
-    "o" => "3",
-    "r" => "0"
-};
+mod consts;
+use consts::*;
 
-const COLORS_3BIT: Map<&str, &str> = phf_map! {
-    "0" => "30",
-    "1" => "34",
-    "2" => "32",
-    "3" => "36",
-    "4" => "31",
-    "5" => "35",
-    "6" => "33",
-    "7" => "37",
-    "8" => "30",
-    "9" => "34",
-    "a" => "32",
-    "b" => "34",
-    "c" => "31",
-    "d" => "35",
-    "e" => "33",
-    "f" => "37",
-    "g" => "33"
-};
+mod depth;
 
-const COLORS_4BIT: Map<&str, &str> = phf_map! {
-    "0" => "30",
-    "1" => "34",
-    "2" => "32",
-    "3" => "36",
-    "4" => "31",
-    "5" => "35",
-    "6" => "33",
-    "7" => "37",
-    "8" => "30",
-    "9" => "34",
-    "a" => "32",
-    "b" => "34",
-    "c" => "31",
-    "d" => "35",
-    "e" => "33",
-    "f" => "37",
-    "g" => "33"
-};
+pub use depth::Depth;
 
-const COLORS_8BIT: Map<&str, &str> = phf_map! {
-    "0" => "0",
-    "1" => "19",
-    "2" => "34",
-    "3" => "37",
-    "4" => "124",
-    "5" => "127",
-    "6" => "214",
-    "7" => "248",
-    "8" => "240",
-    "9" => "147",
-    "a" => "83",
-    "b" => "87",
-    "c" => "203",
-    "d" => "207",
-    "e" => "227",
-    "f" => "15",
-    "g" => "184"
-};
+const ESCAPE_IN_REGEX: [char; 14] = [
+    '[', ']', '(', ')', '{', '}', '*', '+', '.', '$', '^', '\\', '|', '?',
+];
 
-const COLORS_24BIT: Map<&str, [&str; 3]> = phf_map! {
-    "0" => ["0", "0", "0"],
-    "1" => ["0", "0", "170"],
-    "2" => ["0", "170", "0"],
-    "3" => ["0", "170", "170"],
-    "4" => ["170", "0", "0"],
-    "5" => ["170", "0", "170"],
-    "6" => ["255", "170", "0"],
-    "7" => ["170", "170", "170"],
-    "8" => ["85", "85", "85"],
-    "9" => ["85", "85", "255"],
-    "a" => ["85", "255", "85"],
-    "b" => ["85", "255", "255"],
-    "c" => ["255", "85", "85"],
-    "d" => ["255", "85", "255"],
-    "e" => ["255", "255", "85"],
-    "f" => ["255", "255", "255"],
-    "g" => ["221", "214", "5"]
-};
+struct Patterns {
+    codes: Regex,
+    escaped: String,
+}
 
-const FORMAT_TEMPLATES: Map<u8, &str> = phf_map! {
-    3u8 => "\x1b[{}m",
-    4u8 => "\x1b[{}m",
-    8u8 => "\x1b[38;5;{}m",
-    24u8 => "\x1b[38;2;{r};{g};{b}m"
-};
+impl Patterns {
+    pub fn new(marker: char) -> Self {
+        let escaped_marker = ESCAPE_IN_REGEX
+            .contains(&marker)
+            .then(|| format!(r"\{marker}"))
+            .unwrap_or_else(|| marker.to_string());
 
-const BG_FORMAT_TEMPLATES: Map<u8, &str> = phf_map! {
-    3u8 => "\x1b[{}m",
-    4u8 => "\x1b[{}m",
-    8u8 => "\x1b[48;5;{}m",
-    24u8 => "\x1b[48;2;{r};{g};{b}m"
-};
+        let regex = format!("{escaped_marker}(?:{})", *CODE_REGEX);
 
-const COLORS: Map<u8, &Map<&str, &str>> = phf_map! {
-    3u8 => &COLORS_3BIT,
-    4u8 => &COLORS_4BIT,
-    8u8 => &COLORS_8BIT,
-};
+        Self {
+            codes: Regex::new(&regex)
+                .expect("The pattern regex should be valid and properly escaped."),
```

Comment:
for consistency with others
```suggestion
                .expect("the pattern regex should be valid and properly escaped."),
```